bundles/statistics/statsgrid2016/components/Datatable.js

```diff
@@ -376,13 +379,26 @@ Oskari.clazz.define('Oskari.statistics.statsgrid.Datatable', function(sandbox, l
             log.info('Region changed! ', event.getRegionset());
             me._handleRegionsetChanged(event.getRegionset());
         });
+
         this.service.on('StatsGrid.RegionSelectedEvent', function(event) {
             log.info('Region selected! ', event.getRegion());
-            if(me.getCurrentRegionset() !== event.getRegionset()) {
-                // shouldn't be the case ever
-                me._handleRegionsetChanged(event.getRegionset());
-            }
             me.grid.select(event.getRegion());
+
+            var gridEl = me.mainEl.find('table.oskari-grid:visible');
+            var parent = gridEl.parents('.oskari-flyoutcontentcontainer');
+            var compState = state.getComponentState('datatable');
+            if(compState.scroll && parent.length>0) {
+                parent.scrollTop(0);
+                var row = parent.find('tr[data-id="'+event.getRegion()+'"]');
+                if(row.length > 0) {
+                    parent.scrollTop(row.position().top);
```

Comment:
Good point, the scrollable element as second parameter sounds better to me. Let's go with that.